### PR TITLE
use main domains

### PR DESCRIPTION
**Pull request by cbhp**
removed all subdomains (e.g. aahs.luk2.com, ac.luk2.com) using the public suffix list (https://publicsuffix.org/) and replaced them with their main domain (luk2.com).
the check now uses the public suffix list to retrieve the main domain and look it up in the blacklist.

fixes issue #93

### CODE REVIEW

File: lib/spam_email/check.rb [hunk 1]
```diff
@@ -1,11 +1,22 @@
 require 'mail'
+require 'public_suffix'
```
Comment (korny) [inline lib/spam_email/check.rb]: I wonder if we can avoid adding another dependency.

**Comment by korny** [comment]
Very good idea!

Should we always use the main domain and check only once, or is it necessary to do two BLACKLIST checks?

File: lib/spam_email/check.rb (hunk 1 above)
Comment (cbhp) [reply lib/spam_email/check.rb]: The public suffix list is the recommended way to retrieve the main domain. Think about domains like example.com and example.co.uk. Regular expressions and maintaining a list of all possible TLDs is too much effort.

**Comment by cbhp** [comment]
> Very good idea!
> 
> Should we always use the main domain and check only once, or is it necessary to do two BLACKLIST checks?

If the blacklist contains spam.example.org but not example.org and the user registers with badguy@spam.example.org, the lookup has to be performed with the complete domain. 
If the blacklist contains example.org but no subdomains of that and the user registers with badguy@spam.example.org, the lookup has to be performed with the main domain.  
Mostly the second scenario will fit, so we can change the order of the lookups to be more performant.
There can be some domains where I don't want to block the whole domain, but only special subdomains. That's why I implemented two lookups.

**Pull request merged**